desc/equilibrium/equilibrium.py

```diff
@@ -237,12 +248,13 @@ def __init__(
         _assert_nonnegint(N_grid, "N_grid")
 
         self._N = int(setdefault(N, self.surface.N))
-        self._M = int(setdefault(M, self.surface.M))
+        self._M = int(setdefault(M, setdefault(self.surface.M, self.xsection.M)))
         self._L = int(
             setdefault(
                 L,
                 max(
                     self.surface.L,
+                    self.xsection.L,
```

Comment:
i think you can get rid of `self.surface.L` here since its always 0 for a toroidal surface. This was a leftover from when `surface` could also be a cross section